module/modmanager/manager.go

```diff
@@ -1201,10 +1212,12 @@ func (m *module) cleanupAfterStartupFailure(logger logging.Logger) {
 	}
 }
 
-func (m *module) cleanupAfterCrash(mgr *Manager) {
-	if err := m.stopProcess(); err != nil {
-		msg := "Error while stopping process of crashed module"
-		mgr.logger.Errorw(msg, "module", m.cfg.Name, "error", err)
+func (m *module) cleanupAfterCrash(mgr *Manager, processRestarted bool) {
+	if processRestarted {
+		if err := m.stopProcess(); err != nil {
+			msg := "Error while stopping process of crashed module"
+			mgr.logger.Errorw(msg, "module", m.cfg.Name, "error", err)
+		}
 	}
```

Comment:
[minor] since this logic only happens in one place, I'd prefer if we moved this `if` block up to the `defer` statement where `cleanupAfterCrash` is called instead of passing an extra boolean to the function.

Reply:
done